Cabal/src/Distribution/Simple/Command.hs

```diff
@@ -513,6 +529,27 @@ commandsRun globalCommand commands args =
 
      where globalHelp = commandHelp globalCommand
 
+-- Levenshtein distance, from https://wiki.haskell.org/Edit_distance
+-- (Author: JeanPhilippeBernardy, Simple Permissive Licence)
+editDistance :: Eq a => [a] -> [a] -> Int
```

Comment:
This is a bit sad as `cabal-install` already depends on `edit-distance` package. (But adding dependency to `Cabal` is hard).

Reply:
Yeah :-/

I can't think of a better solution though. cabal-install's option parsing is heavily tied to Cabal's

Reply:
Yes. It's not a great design, but that what we have now.